source/validate_id.cpp

```diff
@@ -1216,12 +1216,11 @@ bool idUsage::isValid<SpvOpFunctionParameter>(const spv_instruction_t* inst,
                                    << inst->words[resultTypeIndex]
                                    << "' is not defined.";
```

Comment:
FYI. The spvValidateIDs function in validate.cpp needs to be reimplemented around the spvBinaryParse and spv_parsed_instruction_t because the technique of using spvBinaryOperandInfo function is not expressive enough to describe all the types of instructions, and the fact that they can be variable length.  For example OpSwitch has variable number of operands, and some of them are literal numbers, and others are label Ids. 
I think the entire validate_id.cpp file can be shunk to about 100 lines of code by using a data-driven approach around the spv_parsed_instruction_t.